catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java

```diff
@@ -301,95 +302,131 @@ private boolean orderContainsSupportedDelivery() {
-                        sentFiles.add(currFileName);
-                        currNum++;
+                        try (TemporaryFileBackedOutputStream fos = new TemporaryFileBackedOutputStream(
+                                MAX_MEMORY_SIZE);
+                                GZIPOutputStream zipOut = new GZIPOutputStream(fos)) {
+                            getGzip(zipOut, file.getInputStream());
+                            ByteSource contents = fos.asByteSource();
+                            String currNumPortion = String.format(FILE_COUNT_FORMAT, currNum);
+                            String currFileName =
+                                    filename + "." + currNumPortion + "." + totalNumPortion
+                                            + packagingSpecFormatType.getExtension();
+                            writeFile(destination,
+                                    contents.openStream(),
+                                    contents.size(),
+                                    currFileName,
+                                    packagingSpecFormatType.getContentType());
+                            sentFiles.add(currFileName);
+                            currNum++;
+                        }
                     }
                 }
                 break;
                 case TARUNC: {
-                    ByteSource tar = getTar(files);
-                    String currFileName = filename + packagingSpecFormatType.getExtension();
-                    writeFile(destination,
-                            tar.openStream(),
-                            tar.size(),
-                            currFileName,
-                            packagingSpecFormatType.getContentType());
-                    sentFiles.add(currFileName);
+                    try (TemporaryFileBackedOutputStream fos = new TemporaryFileBackedOutputStream(
+                            MAX_MEMORY_SIZE);
+                            TarOutputStream tarOut = new TarOutputStream(fos)) {
+                        getTar(tarOut, files);
+                        ByteSource tar = fos.asByteSource();
+                        writeFile(destination, packagingSpecFormatType, filename, sentFiles, tar);
+                    }
                 }
                 break;
                 case TARZIP: {
-                    ByteSource tar = getTar(files);
-                    ByteSource zip = getZip(tar.openStream(), filename + ".tar");
-                    String currFileName = filename + packagingSpecFormatType.getExtension();
-                    writeFile(destination,
-                            zip.openStream(),
-                            zip.size(),
-                            currFileName,
-                            packagingSpecFormatType.getContentType());
-                    sentFiles.add(currFileName);
+                    try (TemporaryFileBackedOutputStream fos = new TemporaryFileBackedOutputStream(
```

Comment:
I'd recommend using variable names more descriptive than `fos` and `fos2` when there are nested try-with-resources.